lib/engine/game/g_18_zoo.rb

```diff
@@ -586,6 +586,81 @@ def unowned_purchasable_companies(_entity)
         @available_companies + @future_companies
       end
 
+      def after_par(corporation)
+        bonus_after_par(corporation, 5, 2) if corporation.par_price.price == 9
+        bonus_after_par(corporation, 10, 4) if corporation.par_price.price == 12
+
+        return unless @near_families
+
+        corporations_order = @corporations.sort_by(&:full_name).cycle(2).to_a
+        if @corporations.count(&:ipoed) == 1
+          # Take the first corporation not ipoed after the one just parred
+          next_corporation = corporations_order.drop_while { |c| c.id != corporation.id }
+                                               .find { |c| !c.ipoed }
+          # Take the first corporation not ipoed before the one just parred
+          previous_corporation = corporations_order.reverse
+                                                   .drop_while { |c| c.id != corporation.id }
+                                                   .find { |c| !c.ipoed }
+          @near_families_purchasable = [{ direction: 'next', id: next_corporation.id },
+                                        { direction: 'reverse', id: previous_corporation.id }]
+          @log << "Near family rule: #{previous_corporation.full_name} and #{next_corporation.full_name} are available."
+        else
+          if @corporations.count(&:ipoed) == 2
+            @near_families_direction = @near_families_purchasable.find { |c| c[:id] == corporation.id }[:direction]
+          end
+          corporations = @near_families_direction == 'reverse' ? corporations_order.reverse : corporations_order
+          following_corporation = corporations.drop_while { |c| c.id != corporation.id }
+                                              .find { |c| !c.ipoed }
+          @near_families_purchasable = [{ id: following_corporation.id }]
+
+          @log << "Near family rule: #{following_corporation.full_name} is now available." unless following_corporation
+        end
+      end
+
+      def holiday
+        @holiday ||= company_by_id('HOLIDAY')
+      end
+
+      def midas
+        @midas ||= company_by_id('MIDAS')
+      end
+
+      def too_much_responsibility
+        @too_much_responsibility ||= company_by_id('TOO_MUCH_RESPONSIBILITY')
+      end
+
+      def leprechaun_pot_of_gold
+        @leprechaun_pot_of_gold ||= company_by_id('LEPRECHAUN_POT_OF_GOLD')
+      end
+
+      def it_s_all_greek_to_me
+        @it_s_all_greek_to_me ||= company_by_id('IT_S_ALL_GREEK_TO_ME')
+      end
+
+      def whatsup
+        @whatsup ||= company_by_id('WHATSUP')
+      end
+
+      def apply_custom_ability(company)
+        if company.sym.to_sym == :TOO_MUCH_RESPONSIBILITY
```

Comment:
why do convert these to syms? that's unnecessary

Reply:
ok, but if I don't do it the rspec using this piece of code will fail (because is trying to compare a string with a symbol) :(
any suggestion?

Reply:
why don't you do company.nym == 'TOO_MUCH_RESPONSIBILITY' then?

Reply:
sure, done